storage/local/storage_test.go

```diff
@@ -30,7 +30,92 @@ import (
 )
 
 func TestGetFingerprintsForLabelMatchers(t *testing.T) {
+	storage, closer := NewTestStorage(t)
+	defer closer.Close()
+
+	samples := make([]*clientmodel.Sample, 100)
+	fingerprints := make(clientmodel.Fingerprints, 100)
+
+	for i := range samples {
+		metric := clientmodel.Metric{
+			clientmodel.MetricNameLabel: clientmodel.LabelValue(fmt.Sprintf("test_metric_%d", i)),
+			"label1":                    clientmodel.LabelValue(fmt.Sprintf("test_%d", i/10)),
+			"label2":                    clientmodel.LabelValue(fmt.Sprintf("test_%d", (i+5)/10)),
+		}
+		samples[i] = &clientmodel.Sample{
+			Metric:    metric,
+			Timestamp: clientmodel.Timestamp(i),
+			Value:     clientmodel.SampleValue(i),
+		}
+		fingerprints[i] = metric.Fingerprint()
+	}
+
+	storage.AppendSamples(samples)
+	storage.WaitForIndexing()
+
+	// check retrieves the fingerprints for the given matchers and checks if the
+	// returned fingerprints are the expected ones.
+	check := func(m metric.LabelMatchers, expected clientmodel.Fingerprints) {
+		resfps := storage.GetFingerprintsForLabelMatchers(m)
+		if len(expected) != len(resfps) {
+			t.Fatalf("expected %d matches for %q, found %d", len(expected), m, len(resfps))
+		}
+		for _, fp1 := range resfps {
+			found := false
+			for _, fp2 := range expected {
+				if fp1 == fp2 {
+					found = true
+					break
+				}
+			}
+			if !found {
+				t.Errorf("expected fingerprint %s for %q not in result", fp1, m)
+			}
+		}
+	}
+
+	newMatcher := func(matchType metric.MatchType, name clientmodel.LabelName, value clientmodel.LabelValue) *metric.LabelMatcher {
+		lm, err := metric.NewLabelMatcher(matchType, name, value)
+		if err != nil {
+			t.Fatalf("error creating label matcher: %s", err)
+		}
+		return lm
+	}
+
+	check(metric.LabelMatchers{newMatcher(metric.Equal, "label1", "x")}, fingerprints[:0])
```

Comment:
This seems like a good place to use table-driven tests: https://code.google.com/p/go-wiki/wiki/TableDrivenTests

We are using these for rules tests and in many other places in Prometheus, so it would also be more consistent.

Some examples for orientation:
https://github.com/prometheus/prometheus/blob/master/rules/rules_test.go#L117
https://github.com/prometheus/prometheus/blob/master/config/config_test.go#L24
https://github.com/prometheus/prometheus/blob/master/templates/templates_test.go#L32

What I like about table-driven tests is that inputs and expected outputs are clearly labeled and separated from any common code that executes them. You're almost there with the `check` function, but I think a table would make it look even nicer.